compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -1571,6 +1571,24 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                                 }) {
                                     continue;
                                 }
+
+                                // opaque types generated when desugaring an async function can have a single
+                                // use lifetime even if it is explicitly denied (Issue #77175)
+                                if let hir::Node::Item(item) = self.tcx.hir().get(parent_hir_id) {
+                                    if let hir::ItemKind::OpaqueTy(ref opaque) = item.kind {
+                                        if opaque.origin == hir::OpaqueTyOrigin::AsyncFn {
+                                            // We want to do this only if the liftime identifier is already defined
+                                            // in the async function that generated this. Otherwise it could be
+                                            // an opaque type defined by the developer and we still want this
+                                            // lint to fail compilation
+                                            for p in opaque.generics.params {
```

Comment:
I wonder if it makes sense to build an `FxHashSet` from these to make the amortized cost linear, but that's likely gold-plating.

Reply:
I'm probably missing something. I thought the fx hash would only yield speed gains when fetching a single item. I still have to iterate over the set here.